### PR TITLE
testing/geoclue: upgrade to 2.5.2

### PR DESCRIPTION
**Pull request by maxice8**
- Create user and group geoclue
- Create /var/lib/geoclue with proper permissions
- Switch to meson build system
- Fix license into LGPL-2.1-or-later
- Remove superfluous builddir= assignment and cd "$builddir" statements

### CODE REVIEW

File: testing/geoclue/APKBUILD [hunk 1]
```diff
@@ -1,34 +1,42 @@
...
+license="LGPL-2.1-or-later"
+makedepends="
+	meson
+	libsoup-dev
+	json-glib-dev
+	modemmanager-dev
+	avahi-dev
+	intltool
+	gobject-introspection-dev
+	libnotify-dev"
```
Comment (tcely) [inline testing/geoclue/APKBUILD]:
```suggestion
	libnotify-dev
	"
```

File: testing/geoclue/APKBUILD [hunk 2]
```diff
@@ -1,34 +1,42 @@
...
 subpackages="$pkgname-dev"
-source="https://freedesktop.org/software/geoclue/releases/2.4/geoclue-$pkgver.tar.xz"
-builddir="$srcdir/geoclue-$pkgver"
+install="$pkgname.post-install"
```
Comment (tcely) [inline testing/geoclue/APKBUILD]:
```suggestion
install="$pkgname.pre-install"
```

File: testing/geoclue/APKBUILD (hunk 1 above)
Comment (maxice8) [reply testing/geoclue/APKBUILD]: i prefer keeping the ending quote where it ends.

File: testing/geoclue/geoclue.post-install [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+# Alpine Linux post-install script for socklog
```
Comment (tcely) [inline testing/geoclue/geoclue.post-install]:
```suggestion
# Alpine Linux pre-install script for geoclue
```

File: testing/geoclue/geoclue.post-install [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+# Copyright 2019 Leo (thinkabit.ukim@gmail.com)
+# Distributed under the terms of the GNU General Public License, v2 or later
+
+addgroup -S geoclue 2>/dev/null
+adduser -S -D -H -h /var/lib/geoclue -s /sbin/nologin -G geoclue geoclue 2>/dev/null
+
+install -d -m0755 -o geoclue -g geoclue /var/lib/geoclue
```
Comment (tcely) [inline testing/geoclue/geoclue.post-install]:
```suggestion
install -D -m0755 -o geoclue -g geoclue /var/lib/geoclue
```

File: testing/geoclue/APKBUILD (hunk 2 above)
Comment (maxice8) [reply testing/geoclue/APKBUILD]: They are not required during build-time

File: testing/geoclue/geoclue.post-install (hunk 4 above)
Comment (maxice8) [reply testing/geoclue/geoclue.post-install]: image

File: testing/geoclue/geoclue.post-install (hunk 3 above)
Comment (maxice8) [reply testing/geoclue/geoclue.post-install]: Applied

File: testing/geoclue/APKBUILD (hunk 1 above)
Comment (tcely) [reply testing/geoclue/APKBUILD]: It makes for cleaner diffs when it has a line to itself.

File: testing/geoclue/APKBUILD (hunk 2 above)
Comment (tcely) [reply testing/geoclue/APKBUILD]: How will you set the permissions in the `tar` if they aren't setup when `package` runs?

Comment (maxice8) [reply testing/geoclue/APKBUILD]: There are no geoclue-owned files in the tar, just a post-install script that creates geoclue group and user and installs an empty directory to serve as $HOME for the created geoclue user.

File: testing/geoclue/geoclue.pre-install [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+# Alpine Linux pre-install script for geoclue
+# Copyright 2019 Leo (thinkabit.ukim@gmail.com)
+# Distributed under the terms of the GNU General Public License, v2 or later
+
+addgroup -S geoclue 2>/dev/null
+adduser -S -D -H -h /var/lib/geoclue -s /sbin/nologin -G geoclue geoclue 2>/dev/null
```
Comment (tcely) [inline testing/geoclue/geoclue.pre-install]:
```suggestion
adduser -S -D -h /var/lib/geoclue -s /sbin/nologin -G geoclue geoclue 2>/dev/null
```

File: testing/geoclue/geoclue.pre-install [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+install -d -m0755 -o geoclue -g geoclue /var/lib/geoclue
```
Comment (tcely) [inline testing/geoclue/geoclue.pre-install]:
```suggestion
```

**Pull request closed without merging**